### PR TITLE
Integration test ci

### PR DESCRIPTION
**Pull request by kleash**
- Integration test for installing wasmcc and running sample rust contract
- Updated github actions ci

Fixes issue #4

### CODE REVIEW

**Comment by MHBauer** [comment]
@swetharepakula this looks like it's running. It takes 2 minutes. WDYT?

**Review by MHBauer — APPROVED** [review]
Given the CI looks like it's running, I'm go, but I'd wait for one more review from @swetharepakula if you can hold for a bit.

File: .github/workflows/go.yml
Comment (MHBauer) [inline .github/workflows/go.yml]: maybe not this

**Comment by MHBauer** [comment]
Only thing I'd prefer is to get the local run of this the same as CI. I find it difficult to run locally.

File: .github/workflows/go.yml
Comment (swetharepakula) [inline .github/workflows/go.yml]: Is this piece necessary for ci?

File: .github/workflows/go.yml
Comment (swetharepakula) [inline .github/workflows/go.yml:54]: do you want to use ginkgo here instead since you install it above?

File: .github/workflows/go.yml
Comment (swetharepakula) [inline .github/workflows/go.yml]: You don't need to create the fabric directory

**Comment by swetharepakula** [comment]
So I think you how it is right now is fine, but I agree with Morgan. Perhaps it is useful to make a script for some of the intermediary steps so that this can be recreated locally as well.

File: .github/workflows/go.yml
Comment (kleash) [reply .github/workflows/go.yml]: Nope, just added it for more logs. I will remove it

File: .github/workflows/go.yml
Comment (kleash) [reply .github/workflows/go.yml]: correct, thanks for pointing out

File: .github/workflows/go.yml
Comment (kleash) [reply .github/workflows/go.yml]: removed pwd

File: .github/workflows/go.yml
Comment (kleash) [reply .github/workflows/go.yml:54]: go bin is not in path, that's why I avoided it. Is there any advantage of using ginkgo over go test?

Comment (kleash) [reply .github/workflows/go.yml:54]: plus it needs gcc: https://github.com/kleash/fabric-chaincode-wasm/runs/268514684

**Comment by kleash** [comment]
> So I think you how it is right now is fine, but I agree with Morgan. Perhaps it is useful to make a script for some of the intermediary steps so that this can be recreated locally as well.

Ok. The only main thing required to run this is Fabric docker images and HLF 1.4.0 in go path. Maybe I can add both these in a script?

File: .github/workflows/go.yml
Comment (swetharepakula) [reply .github/workflows/go.yml:54]: No real advantage, just curious as to why you were installing it above.

Comment (kleash) [reply .github/workflows/go.yml:54]: Oh ok. I think I tried using ginkgo first and later replaced it with go test. But didn't remove the installation part.

I have removed ginkgo installation now. You are right, it's an extraneous step.

File: integration/e2e/utils.go
Comment (swetharepakula) [inline integration/e2e/utils.go]: You don't need this struct and its corresponding methods. We needed it so we could get hex encoded output. So you can remove that piece.

File: integration/e2e/utils.go
Comment (swetharepakula) [inline integration/e2e/utils.go]: I added this function because we needed to build fab3. Since you don't, you could directly just call components.Build(). You don't need this extra helper function.

File: integration/e2e/e2e_test.go
Comment (swetharepakula) [inline integration/e2e/e2e_test.go]: missing the `y` in Query

File: integration/e2e/e2e_test.go
Comment (swetharepakula) [inline integration/e2e/e2e_test.go]: Missing the `y` in Query

File: integration/e2e/e2e_test.go
Comment (swetharepakula) [inline integration/e2e/e2e_test.go]: E2E or e2e

**Review by swetharepakula — APPROVED** [review]
There are a few nits & typos. I think there are some additional functions in your utils.go that you don't need. I don't mind merging as is and having follow up prs to fix the smaller things.

File: integration/e2e/utils.go
Comment (kleash) [reply integration/e2e/utils.go]: fixed

File: integration/e2e/utils.go
Comment (kleash) [reply integration/e2e/utils.go]: removed it

File: integration/e2e/e2e_test.go
Comment (kleash) [reply integration/e2e/e2e_test.go]: fixed. Thanks for pointing out

File: integration/e2e/e2e_test.go
Comment (kleash) [reply integration/e2e/e2e_test.go]: fixed. Thanks for pointing out

File: integration/e2e/e2e_test.go
Comment (kleash) [reply integration/e2e/e2e_test.go]: fixed

**Comment by swetharepakula** [comment]
@kleash Rebase, and update the branch. Then we can go ahead and merge this.

**Review by swetharepakula — APPROVED** [review]

**Pull request merged**
